### PR TITLE
Command.Deploy: add checkSystemKnownHosts option to ob deploy init

### PR DESCRIPTION
**Pull request by jmininger**
Command.Deploy: add checkSystemKnownHosts option to ob deploy init

This change gives users the option to allow the 'deploy init' command to check their known_hosts file for any host-key pairs matching the hosts passed in and put them in the configuration's known_hosts files. Any host previously acknowledged as 'recognized by the system can then connect without prompt during ob init.

Addreses issue #513

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -206,7 +206,7 @@ ob deploy init \
...
...loy init` to check your `~/.ssh/known_hosts` file and save any fingerprints matching the host to the deployment-specific configuration by passing the `check_known_hosts` option to the `deploy init` command. **Obelisk deployments do *not* rely on the `known_hosts` of your local machine during deployment.** This is because, in the event that you need to switch from one deploy machine / bastion host to another, you want to be absolutely sure that you're still connecting to the machines you think you are, even if that deploy machine / bastion host has never connected to them before. Obelisk explicitly avoids a workflow that encourages people to accept host keys without checking them, since that could result in leaking production secrets to anyone who manages to MITM you, e.g. via DNS spoofing or cache poisoning. (Note that an active attack is a circumstance where you may need to quickly switch bastion hosts, e.g. because the attacker has taken one down or you have taken it down in case it was compromised. In this circumstance you might need to deploy to production to fix an exploit or rotate keys, etc.) When you run `ob deploy` later it will rely on the saved verification in this step.
```
Comment (JBetz) [inline README.md]: Shouldn't it be `use-known-hosts` rather than `check_known_hosts`?

File: lib/command/src/Obelisk/Command.hs [hunk 2]
```diff
@@ -203,6 +204,7 @@ data DeployInitOpts = DeployInitOpts
   , _deployInitOpts_route :: String
   , _deployInitOpts_adminEmail :: String
   , _deployInitOpts_enableHttps :: Bool
+  , _deployInitOpts_checkSystemKnownHosts :: Bool
```
Comment (JBetz) [inline lib/command/src/Obelisk/Command.hs]: Same thing here, looks like a rename didn't get propagated everywhere.

File: README.md (hunk 1 above)
Comment (3noch) [reply README.md]: `check-known-hosts` but yeah

File: lib/command/src/Obelisk/Command/Deploy.hs [hunk 3]
```diff
@@ -295,6 +302,24 @@ readDeployConfig :: MonadObelisk m => FilePath -> FilePath -> m String
 readDeployConfig deployDir fname = liftIO $ do
   fmap (T.unpack . T.strip) $ T.readFile $ deployDir </> fname
 
+lookupKnownHosts :: MonadObelisk m  => String -> m [BS.ByteString]
+lookupKnownHosts hostName =
+  fmap filterComments $ readCreateProcessWithExitCode $ proc "ssh-keygen" ["-F", hostName]
```
Comment (3noch) [inline lib/command/src/Obelisk/Command/Deploy.hs]: We'll need to either `staticWhich` this. But do we *need* to use the host's version of this? Thinking...

**Comment by romefeller** [comment]
Moving it to: https://github.com/obsidiansystems/obelisk/pull/916

**Pull request closed without merging**